akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -753,27 +756,39 @@ final class Partition[T](val outputPorts: Int, val partitioner: T ⇒ Int, val e
   override val shape: UniformFanOutShape[T, T] = UniformFanOutShape[T, T](in, out: _*)
 
   override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler {
+    val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
     private var outPendingElem: Any = null
     private var outPendingIdx: Int = _
     private var downstreamRunning = outputPorts
 
     def onPush() = {
       val elem = grab(in)
-      val idx = partitioner(elem)
-      if (idx < 0 || idx >= outputPorts) {
-        failStage(PartitionOutOfBoundsException(s"partitioner must return an index in the range [0,${outputPorts - 1}]. returned: [$idx] for input [${elem.getClass.getName}]."))
-      } else if (!isClosed(out(idx))) {
-        if (isAvailable(out(idx))) {
-          push(out(idx), elem)
-          if (out.exists(isAvailable(_)))
+      try {
+        val idx = partitioner(elem)
+        if (idx < 0 || idx >= outputPorts) {
+          failStage(PartitionOutOfBoundsException(s"partitioner must return an index in the range [0,${outputPorts - 1}]. returned: [$idx] for input [${elem.getClass.getName}]."))
```

Comment:
should we support supervision for this also? then it should throw instead of `failStage`

Reply:
This should be changed to throwing @yoshiyoshifujii

Reply:
@yoshiyoshifujii I think this one was missed. We want to be able to supervise `PartitionOutOfBoundsException`. Now this isn't in the throw it could just call the decide directly.